packages/debugger/lib/data/selectors/index.js

```diff
@@ -167,43 +168,142 @@ const data = createSelectorTree({
+                  .slice()
+                  .reverse();
+                linearizedBaseContractsFromBase.pop(); //remove the last element, i.e.,
+                //the contract itself, because we want to treat that one specially
+                //now, we put it all together
+                newScope.variables = []
+                  .concat(
+                    //concatenate the variables lists from the base classes
+                    ...linearizedBaseContractsFromBase.map(
+                      contractId =>
+                        scopes[compilation][contractId].variables || []
+                      //we need the || [] because contracts with no state variables
+                      //have variables undefined rather than empty like you'd expect
+                    )
+                  )
+                  .filter(
+                    variable =>
+                      inlined[compilation][variable.id].definition
+                        .visibility !== "private"
+                    //filter out private variables from the base classes
+                  )
+                  //add in the variables for the contract itself -- note that here
+                  //private variables are not filtered out!
+                  .concat(scopes[compilation][id].variables || [])
+                  .filter(variable => {
+                    //HACK: let's filter out those constants we don't know
+                    //how to read.  they'll just clutter things up.
+                    debug("variable %O", variable);
+                    let definition =
+                      inlined[compilation][variable.id].definition;
+                    return (
+                      !definition.constant ||
+                      Codec.Ast.Utils.isSimpleConstant(definition.value)
+                    );
+                  });
+
+                return { [id]: newScope };
+              })
+            )
+          }))
+        )
+      ),
+
+      /*
+       * data.views.scopes.raw
+       * NOTE: grouped by compilation
+       *
+       * the raw scopes data, just with intermediate
+       * layers cut out
+       */
+      raw: createLeaf(["/info/scopes"], scopes =>
+        Object.assign(
+          {},
+          ...Object.entries(scopes.byCompilation).map(
+            ([compilation, { byId: nodes }]) => ({
+              [compilation]: nodes
+            })
+          )
+        )
+      ),
+
       /**
        * data.views.scopes.inlined (namespace)
        */
       inlined: {
         /**
          * data.views.scopes.inlined (selector)
-         * see data.info.scopes for how this differs from the raw version
+         * see data.views.scopes for how this differs from the raw version
+         * NOTE: grouped by compilation
          */
-        _: createLeaf(["/info/scopes", "./raw"], (scopes, inlined) =>
+        _: createLeaf(["../_", "./raw"], (scopes, inlined) =>
```

Comment:
do you need to do `../_`? I would expect just `..` to work, but I forget the implementation.

Reply:
I dunno.  I ran into a bunch of problems with this, this is what ultimately worked.  You're right I didn't try just `..` though.  Can try it and see what happens.  Might want to leave as `../_` for clarity regardless though.

Reply:
yeah you're probably right!